source/patterns/@aws-solutions-constructs/aws-lambda-opensearch/README.md

```diff
@@ -0,0 +1,164 @@
+# aws-lambda-opensearch module
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> All classes are under active development and subject to non-backward compatible changes or removal in any
+> future version. These are not subject to the [Semantic Versioning](https://semver.org/) model.
+> This means that while you may use them, you may need to update your source code when upgrading to a newer version of this package.
+
+---
+<!--END STABILITY BANNER-->
+
+| **Reference Documentation**:| <span style="font-weight: normal">https://docs.aws.amazon.com/solutions/latest/constructs/</span>|
+|:-------------|:-------------|
+<div style="height:8px"></div>
+
+| **Language**     | **Package**        |
+|:-------------|-----------------|
+|![Python Logo](https://docs.aws.amazon.com/cdk/api/latest/img/python32.png) Python|`aws_solutions_constructs.aws_lambda_opensearch`|
+|![Typescript Logo](https://docs.aws.amazon.com/cdk/api/latest/img/typescript32.png) Typescript|`@aws-solutions-constructs/aws-lambda-opensearch`|
+|![Java Logo](https://docs.aws.amazon.com/cdk/api/latest/img/java32.png) Java|`software.amazon.awsconstructs.services.lambdaopensearch`|
+
+## Overview
+This AWS Solutions Construct implements the AWS Lambda function and Amazon OpenSearch Service with the least privileged permissions.
+
+**Some cluster configurations (e.g VPC access) require the existence of the `AWSServiceRoleForAmazonOpenSearchService` Service-Linked Role in your account.**
+
+**You will need to create the service-linked role using the AWS CLI once in any account using this construct (it may have already been executed to support other stacks):**
+```
+aws iam create-service-linked-role --aws-service-name es.amazonaws.com
```

Comment:
This is the same command as the Elasticsearch page - are we sure about that? Is there a different command for the opensearch role?

Reply:
Confirmed same command

Reply:
I think we can drop this section - this creates a the legacy ElasticSearch role, but there is a new role for OpenSearch which appears to be functionally identical. If an account has the ElasticSearch role then the docs say that's sufficient, but if they don't then creating an OpenSearch domain will _automatically_ create the OpenSearch role.

Docs [here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/slr.html):

> **Legacy Elasticsearch service-linked role**
> Amazon OpenSearch Service uses a service-linked role called AWSServiceRoleForAmazonOpenSearchService. Your accounts might also contain a legacy service-linked role called AWSServiceRoleForAmazonElasticsearchService, which works with the deprecated Elasticsearch API endpoints.
> 
> If the legacy Elasticsearch role doesn't exist in your account, OpenSearch Service automatically creates a new OpenSearch service-linked role the first time you create an OpenSearch domain. Otherwise your account continues to use the Elasticsearch role. In order for this automatic creation to succeed, you must have permissions for the iam:CreateServiceLinkedRole action.